pkg/apis/crd.nsx.vmware.com/v1alpha1/ipblockinfo_types.go

```diff
@@ -0,0 +1,42 @@
+/* Copyright © 2024 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:resource:scope="Cluster"
+
+// IPBlockInfo is the Schema for the ipblockinfo API
+type IPBlockInfo struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	CIDRs []CIDR `json:"cidrs"`
```

Comment:
Maybe easier for consumers to define two separate fields - ExternalIPBlocks and PrivateTGWIPBlocks.

Reply:
Updated, I also added the PrivateIPBlocks. But if we store the ipblocks with different visibility in different arrays, do we still need to create the IPBlockInfo struct or we can just save cidrs as string array in the corresponding ipblocks array?